sway/tree/view.c

```diff
@@ -73,7 +74,46 @@ void view_set_activated(struct sway_view *view, bool activated) {
 	}
 }
 
+void view_set_fullscreen(struct sway_view *view, bool fullscreen) {
```

Comment:
This function should work when the view is unmapped, i.e. when not visible on screen. In this case the view doesn't have a container yet.

Test case: launch mpv with the fullscreen flag.

Reply:
I don't think this function is called before the view is mapped, so that's not a problem.

I tested mpv with the fullscreen argument and realised that I need to adjust the `handle_map()` functions to check if the surface is requesting fullscreen while mapping. I've done this and mpv's fullscreen argument works now.

Reply:
>I don't think this function is called before the view is mapped, so that's not a problem.

Are you sure about this? It seems odd to me. Are you starting mpv with the Wayland backend?

Reply:
bump

Reply:
Oh, you need the not-yet-released mpv (from Git), this is annoying. Alternatively, you can use my test client here: https://gist.github.com/emersion/34ed72adecd78bdfac9208a97ffb46a3

Reply:
Resolved in 2acfbc5.